### DISCUSSION TITLE
Add button to show CMS page

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#41082] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

There is no button in CMS pages section in backoffice to preview a CMS page. Only when clicking Save and Preview (which gets blocked in Chrome by default because it looks like an unwanted popup).

If you don't have this page added to a linklist module for example, it's very hard to get there in the front, you need to construct the URL manually. :-)

### Describe the solution you'd like

A "Preview" button, a magnifying glass, should be added to every line in **CMS list** page.
![Snímek obrazovky 2022-03-30 132001](https://user-images.githubusercontent.com/6097524/160823153-bbcc6579-bf32-43ec-aaa7-3d4055c54ea8.png)

A button should be added to **CMS edit** page.
![view](https://user-images.githubusercontent.com/6097524/160823343-36c675dc-a9ff-4359-803b-b23801a0288f.png)

### Additional context

Ping @MatShir @kpodemski for specs

### Do you plan to work on this feature?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. 
So, we're waiting for your PR :rocket: and thanks for the proposed design :muscle: 

Thank you

**Comment by MatShir** [comment]
Seems a great idea ! let's go !

**Comment by MatShir** [comment]
The button could be called 'Preview' to stay consistent

**Comment by kpodemski** [comment]
Hey @Hlavtox 

Do you want me to assign you to this issue? :)

**Comment by Hlavtox** [comment]
@MatShir @kpodemski OK, let's do it! :-) 

**Few things to clarify:**
- Which language it should open in? The employee default one like on product page?
- About multistore:
  - If all shops context is selected, it will be previewed on a default store.
  - If a single shop context is selected, it will be previewed on the selected store.

**Comment by kpodemski** [comment]
@Hlavtox 

Sounds good. I think we can easily change that if needed, but your idea makes the most sense.

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Could you please add/modify your PR following this new proposed design:

![image](https://user-images.githubusercontent.com/23308427/190982996-d35d1028-2de8-4e35-a7e8-bc53e39feec9.png)

Ping @PrestaShop/ux-team could you please check and approve it :pray: 
FYI: @bullmade :wink: 

Thanks!

**Comment by ElodieOS** [comment]
Hello @hibatallahAouadni ! Yes, this is a great idea. 

Normally, we would like to have only two icons in the action column of the tables, but the preview can be an exception, as it's more comfortable for the merchants to see it right away. Besides, the icon is very understandable.

The duplicate action is also interesting for merchants, so it's a good idea to add it to the options submenu (more_vertical icon). 

Also, I noticed that the "save and preview" button in a CMS page still didn't work (maybe it's the Chrome issue mentioned above).

**It's validated** 👍

**Comment by bullmade** [comment]
@ElodieOS can you please reconsider the "Save and preview" feature. Could it be "Save and stay" instead, and then make Preview as a seperate button.

It dosen't work very well intentioned if a merchant wants to see their changes frequently while they edit, and a new window constantly opens up!
Save the merchant some clicks, and let them have "Save and stay" and a preview button, and after they press "save and stay", they can simply go to their second tab and refresh.

**Comment by bullmade** [comment]
And speaking of "Save and stay". It's missing on the category editpage too (in my opinion).

**Comment by Hlavtox** [comment]
@bullmade @ElodieOS PR ready https://github.com/PrestaShop/PrestaShop/pull/29678

**Comment by Hlavtox** [comment]
@bullmade Modified both "Save" buttons to stay on the edit page after saving. ;-)

**Comment by Hlavtox** [comment]
Still relevant
